posthog/hogql_queries/insights/trends/breakdown.py

```diff
@@ -0,0 +1,118 @@
+from typing import List, Tuple
+from posthog.hogql import ast
+from posthog.hogql_queries.insights.trends.breakdown_values import BreakdownValues
+from posthog.hogql_queries.insights.trends.utils import series_event_name
+from posthog.hogql_queries.utils.query_date_range import QueryDateRange
+from posthog.models.filters.mixins.utils import cached_property
+from posthog.models.team.team import Team
+from posthog.schema import ActionsNode, EventsNode, TrendsQuery
+
+
+class Breakdown:
+    query: TrendsQuery
+    team: Team
+    series: EventsNode | ActionsNode
+    query_date_range: QueryDateRange
+
+    def __init__(
+        self, team: Team, query: TrendsQuery, series: EventsNode | ActionsNode, query_date_range: QueryDateRange
+    ):
+        self.team = team
+        self.query = query
+        self.series = series
+        self.query_date_range = query_date_range
+
+    @cached_property
+    def enabled(self):
+        return self.query.breakdown is not None and self.query.breakdown.breakdown is not None
+
+    @cached_property
+    def is_histogram_breakdown(self):
+        return self.enabled and self.query.breakdown.breakdown_histogram_bin_count is not None
+
+    def placeholders(self):
+        values = self._get_breakdown_buckets_ast() if self.is_histogram_breakdown else self._get_breakdown_values_ast
+
+        return {"cross_join_breakdown_values": ast.Alias(alias="breakdown_value", expr=values)}
+
+    def events_select(self):
+        if self.is_histogram_breakdown:
+            return ast.Alias(alias="breakdown_value", expr=self._get_breakdown_histogram_multi_if())
+
+        return ast.Alias(alias="breakdown_value", expr=ast.Field(chain=["properties", self.query.breakdown.breakdown]))
+
+    def events_where_filter(self):
+        return ast.CompareOperation(
+            left=ast.Field(chain=["properties", self.query.breakdown.breakdown]),
+            op=ast.CompareOperationOp.In,
+            right=self._get_breakdown_values_ast,
+        )
+
+    def _get_breakdown_buckets_ast(self) -> ast.Array:
+        buckets = self._get_breakdown_histogram_buckets()
+        values = list(map(lambda t: f"[{t[0]},{t[1]}]", buckets))
+        values.append('["",""]')
+
+        return ast.Array(exprs=list(map(lambda v: ast.Constant(value=v), values)))
+
+    @cached_property
+    def _get_breakdown_values_ast(self) -> ast.Array:
+        return ast.Array(exprs=list(map(lambda v: ast.Constant(value=v), self._get_breakdown_values)))
```

Comment:
I've usually seen `map` written as a inline list loop in Python:

```suggestion
        return ast.Array(exprs=[ast.Constant(value=v) for v in self._breakdown_values])
```